pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java

```diff
@@ -43,7 +68,7 @@ static boolean castToBoolean(final Object in) throws PSQLException {
       return fromString((String) in);
     }
     if (in instanceof Character) {
-      return fromCharacter((Character) in);
+      return fromString(in.toString());
```

Comment:
Well, one of the options is just to keep things as is. That would prevent performance regressions. `fromCharacter` was not that hard to maintain, was it?